openmdao/docs/openmdao_book/features/building_blocks/components/ks_comp.ipynb

```diff
@@ -394,6 +394,150 @@
    "source": [
     "assert_near_equal(prob.get_val('ks.KS', indices=0), np.amax(prob.get_val('x')), tolerance=1e-8)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Computing Min and Max of KS**\n",
+    "\n",
+    "Because KSComp is a smooth max function, you can pass in any values to \"g\" and it will approximate the min and max. \n",
+    "\n",
+    "Here we set an upper limit which is lower than our \"g\" input and `lower_flag` to True to find the maximum value."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "n = 50\n",
+    "prob = om.Problem()\n",
+    "model = prob.model\n",
+    "\n",
+    "prob.driver = om.ScipyOptimizeDriver()\n",
+    "\n",
+    "model.add_subsystem('comp', om.ExecComp('y = -3.0*x**2 + k',\n",
+    "                                        x=np.zeros((n, )),\n",
+    "                                        y=np.zeros((n, )),\n",
+    "                                        k=0.0), promotes_inputs=['x', 'k'])\n",
+    "\n",
+    "model.add_subsystem('ks', om.KSComp(width=n, upper=1.0))\n",
+    "\n",
+    "model.add_design_var('k', lower=-10, upper=10)\n",
+    "model.add_objective('k', scaler=-1)\n",
+    "model.ks.options['lower_flag'] = True\n",
+    "\n",
+    "model.connect('comp.y', 'ks.g')\n",
+    "\n",
+    "prob.setup()\n",
+    "prob.set_val('x', np.linspace(-np.pi/2, np.pi/2, n))\n",
+    "prob.set_val('k', 5.)\n",
+    "\n",
+    "prob.run_driver()\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "\n",
+    "x = prob.get_val('x')\n",
+    "y = prob.get_val('comp.y')\n",
+    "\n",
+    "ax.plot(x, y, 'r.')\n",
+    "ax.set_xlabel('x')\n",
+    "ax.set_ylabel('y')\n",
+    "ax.grid(True)\n",
+    "ax.text(-0.25, 0, f\"k = {prob.get_val('k')[0]:6.3f}\")\n",
+    "\n",
+    "plt.show()\n",
+    "\n",
+    "print(prob['ks.KS'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "remove-output",
+     "remove-input"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "assert(prob['ks.KS'] <= 0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we set an upper limit which is higher than our \"g\" input to find the minimum value."
```

Comment:
Suggested text:

**Example: KSComp for aggregating a constraint with a non-zero upper bound. 

Here we use the KSComp to aggregate a constraint with an upper bound.